### PR TITLE
Build for mac osx

### PR DESCRIPTION
**Pull request by pkernevez**
The modification I had to do to build the project for mac osx

Thanks for you project

### CODE REVIEW

File: slsqp4j/build.gradle [hunk 1]
```diff
@@ -25,8 +25,10 @@ task compileC(type: Exec) {
 
     group "Build"
     description "Compiles the C sources."
-    commandLine 'gcc', '-fPIC', '-c', '-o', obj, '-I', System.properties['java.home'] + "/include",
-        '-I', System.properties['java.home'] + "/include/linux", src
+    logger.warn("Java home " + System.properties['java.home'] )
+    commandLine 'gcc-10', '-fPIC', '-c', '-o', obj, '-I', "/Library/Java/JavaVirtualMachines/jdk8/Contents/Home" + "/include",
```
Comment (jamesasefa) [inline slsqp4j/build.gradle:29]: Could you wrap this in an if statement that checks the current OS. Can be done via:
```
    OperatingSystem os = org.gradle.nativeplatform.platform.internal.DefaultNativePlatform.currentOperatingSystem;
    if (os.isLinux()) {
        commandLine 'gcc', '-fPIC', '-c', '-o', obj, '-I', System.properties['java.home'] + "/include",
                '-I', System.properties['java.home'] + "/include/linux", src
    } else if (os.isMacOsX()) {
        commandLine 'gcc-10', '-fPIC', '-c', '-o', obj, '-I', "/Library/Java/JavaVirtualMachines/jdk8/Contents/Home" + "/include",
                '-I', "/Library/Java/JavaVirtualMachines/jdk8/Contents/Home" + "/include/darwin", src
    }
```

**Review by jamesasefa — CHANGES_REQUESTED** [review]

**Comment by jamesasefa** [comment]
Hi - thanks very much for your contribution. I'm glad you've found the project useful! I requested a change but in addition, could you update the `.gitignore` file to ignore the compiled shared object. I think it's best to remove the shared objects from the project entirely and have users build it themselves.

File: slsqp4j/build.gradle (hunk 1 above)
Comment (pkernevez) [reply slsqp4j/build.gradle]: Should be good now.
I replayed the whole installation on BigSur and Mojave OSX version.

**Comment by pkernevez** [comment]
Hi,

Thanks for your project, it's very useful for us.
The best thing for users should be to publish artefacts with os classifier extensions, using a plugin like this one https://github.com/trustin/os-maven-plugin (I don't know the right one to use with gradle).

**Review by jamesasefa — APPROVED** [review]

**Pull request merged**